### ISSUE TITLE
SchedulerStarted before StartPlugins leads to System.TimeZoneNotFoundException with 

### ISSUE THREAD

**Issue by klinki** [quartznet/quartznet#831]
### Version: 3.0.7
 

OS: Windows 10 (I use Windows for development, Linux for production usage).

### Expected behavior
`Quartz.Plugins.TimeZoneConverter` should be used when installed and should be respected in `SchedulerStarted` method.  `System.TimeZoneNotFoundException` should not be thrown when starting scheduler.

### Actual behavior
`Quartz.Plugins.TimeZoneConverter` is not properly initialized so it is not used and `SchedulerStarted` method throws `System.TimeZoneNotFoundException`.

I believe in `QuartzScheduler.Start`
```
await StartPlugins(cancellationToken).ConfigureAwait(false);
```
should be run before 
```
await resources.JobStore.SchedulerStarted(cancellationToken).ConfigureAwait(false);
```
to allow initialization of Quartz.Plugins.TimeZoneConverter.

### Steps to reproduce

Scheduler configuration:
```json
  "quartz": {
    "quartz.serializer.type": "json",
    "quartz.threadPool.threadCount": "2",
    "quartz.jobStore.type": "Quartz.Impl.AdoJobStore.JobStoreTX, Quartz",
    "quartz.jobStore.useProperties": "true",
    "quartz.jobStore.tablePrefix": "__QRTZ_",
    "quartz.jobStore.dataSource": "mysql",
    "quartz.dataSource.mysql.provider": "MySql",
    "quartz.dataSource.mysql.connectionString": "....",
    "quartz.plugin.timezoneConverter.type": "Quartz.Plugin.TimeZoneConverter.TimeZoneConverterPlugin, Quartz.Plugins.TimeZoneConverter"
  },
```

```csharp
// scheduler and job configuration, SystemTime prefereably set to fixed point
```

Full exception:
```
2020-03-08 10:55:00.060 +01:00 [ERR] Failure occurred during job recovery: Couldn't retrieve trigger: Could not find time zone with id Europe/Budapest, consider using Quartz.Plugins.TimeZoneConverter for resolving more time zones ids
2020-03-08 10:55:00.060 +01:00 [ERR] Error while processing job
Quartz.JobPersistenceException: Couldn't retrieve trigger: Could not find time zone with id Europe/Budapest, consider using Quartz.Plugins.TimeZoneConverter for resolving more time zones ids
 ---> System.TimeZoneNotFoundException: Could not find time zone with id Europe/Budapest, consider using Quartz.Plugins.TimeZoneConverter for resolving more time zones ids
 ---> System.TimeZoneNotFoundException: The time zone ID 'Europe/Budapest' was not found on the local computer.
   at System.TimeZoneInfo.FindSystemTimeZoneById(String id)
   at Quartz.Util.TimeZoneUtil.FindTimeZoneById(String id)
   --- End of inner exception stack trace ---
   at Quartz.Util.TimeZoneUtil.FindTimeZoneById(String id)
   at Quartz.Impl.AdoJobStore.CronTriggerPersistenceDelegate.LoadExtendedTriggerProperties(ConnectionAndTransactionHolder conn, TriggerKey triggerKey, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.StdAdoDelegate.SelectTrigger(ConnectionAndTransactionHolder conn, TriggerKey triggerKey, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.RetrieveTrigger(ConnectionAndTransactionHolder conn, TriggerKey triggerKey, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Quartz.Impl.AdoJobStore.JobStoreSupport.RetrieveTrigger(ConnectionAndTransactionHolder conn, TriggerKey triggerKey, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.RecoverMisfiredJobs(ConnectionAndTransactionHolder conn, Boolean recovering, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.RecoverJobs(ConnectionAndTransactionHolder conn, CancellationToken cancellationToken)
   at Quartz.Impl.AdoJobStore.JobStoreSupport.<>c__DisplayClass260_0.<<ExecuteInNonManagedTXLock>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Quartz.Impl.AdoJobStore.JobStoreSupport.ExecuteInNonManagedTXLock[T](String lockName, Func`2 txCallback, Func`3 txValidator, CancellationToken cancellationToken)
```

**Comment by lahma** [comment]
Thank you for bringing this up and having the details to understand the use case. In hindsight the plugin indeed is initialized too late. There shouldn't be anything breaking if initialization order is changed. A PR would be most welcome too 🙂

**Comment by klinki** [comment]
Great, thanks :) PR is ready.

**Comment by lahma** [comment]
This was released as part of 3.1 beta 1 so closing, thanks again.
